### PR TITLE
Reworked the AMO Dump to run completely in memory

### PR DESCRIPTION
**Pull request by crankycoder**
This reworks the taar_amodump job to run completely in memory and not write any temporary files to disk.

cc/ @acmiyaguchi

### CODE REVIEW

File: mozetl/taar/taar_amodump.py [hunk 1]
```diff
@@ -104,130 +73,113 @@ class AMOAddonInfo(JSONSchema):
...
+                final_result[k] = v
+        logger.info("Final addon set includes %d addons." % len(final_result))
+        return final_result
+
+    def _fetch_pages(self):
+        addon_map = {}
+
+        urls = []
+        for i in range(1, self._page_count+1):
+            url = "%s%s%s" % (DEFAULT_AMO_REQUEST_URI, QUERY_PARAMS, ("&page=%d" % i))
```
Comment (acmiyaguchi) [inline mozetl/taar/taar_amodump.py]: I know this existed in the previous version too, so not a big deal. String formatting would look better here: 

```python
url = "{}{}{}&page={}".format(...)
```

File: mozetl/taar/taar_amodump.py [hunk 2]
```diff
@@ -324,49 +277,24 @@ def marshal(value, name, type_def):
...
+    try:
+        store_json_to_s3(json.dumps(addon_map),
                          AMO_DUMP_FILENAME,
                          date,
                          s3_prefix,
                          s3_bucket)
-    finally:
-        logger.info("Cleaning up and removing : %s" % path)
-        shutil.rmtree(path, ignore_errors=True)
+        logger.info("Completed uploading s3://%s/%s%s.json" % (s3_bucket, s3_prefix, AMO_DUMP_FILENAME))
```
Comment (acmiyaguchi) [inline mozetl/taar/taar_amodump.py]: Looks like there's a single linting error on this line.

**Comment by crankycoder** [comment]
thanks @acmiyaguchi !  I've updated the PR as per your review.

**Comment by acmiyaguchi** [comment]
@crankycoder merging this in, let me know if you have any issues related to deployment.

**Pull request merged**